### PR TITLE
Release 4.3.0

### PR DESCRIPTION
**Pull request by RiteshHMCTS**
### Change description ###
merged master to release-4.3.0

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by udaydenduluri33 — DISMISSED** [review]
@RiteshHMCTS I can see the changes from master going to Release

File: src/shared/components/palette/datetime-picker/datetime-picker.component.ts [hunk 1]
```diff
@@ -127,7 +127,7 @@ export class DatetimePickerComponent extends AbstractFormFieldComponent implemen
       this.hideTime = true;
       this.checkTime = false;
       this.dateTimeEntryFormat = this.formatTranslationService.removeTime(this.dateTimeEntryFormat);
-      this.momentFormat = 'YYYY-MM-DD';
+      this.momentFormat = 'YYYY-MM-DD'
```
Comment (rayliangatos) [inline src/shared/components/palette/datetime-picker/datetime-picker.component.ts:130]: why ; is removed

File: src/app.config.ts [hunk 2]
```diff
@@ -4,6 +4,12 @@ export abstract class AbstractAppConfig {
   abstract getApiUrl(): string;
   abstract getCaseDataUrl(): string;
   abstract getDocumentManagementUrl(): string;
+  getDocumentManagementUrlV2(): string {
+    return undefined;
```
Comment (rayliangatos) [inline src/app.config.ts:8]: what's the point return the undefined

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 3]
```diff
@@ -207,49 +209,79 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
+    this.uploadedDocument = this.registerControl(new FormGroup(documentFormGroup), true) as FormGroup;
   }
 
-  private createDocumentForm(url: string, binaryUrl: string, filename: string) {
-    this.uploadedDocument = this.registerControl(new FormGroup({
-      document_url: new FormControl(url),
-      document_binary_url: new FormControl(binaryUrl),
-      document_filename: new FormControl(filename)
-    }), true) as FormGroup;
+  private createDocumentForm(document: FormDocument) {
```
Comment (rayliangatos) [inline src/shared/components/palette/document/write-document-field.component.ts:272]: not return type

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 4]
```diff
@@ -207,49 +209,79 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
+    }
   }
-  private createDocumentFormWithValidator(url: string, binaryUrl: string, filename: string) {
-    this.uploadedDocument = this.registerControl(new FormGroup({
-      document_url: new FormControl(url, Validators.required),
-      document_binary_url: new FormControl(binaryUrl, Validators.required),
-      document_filename: new FormControl(filename, Validators.required)
-    }), true) as FormGroup;
+
+  private createDocumentFormWithValidator(document: FormDocument) {
```
Comment (rayliangatos) [inline src/shared/components/palette/document/write-document-field.component.ts:257]: no return type

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 5]
```diff
@@ -100,19 +120,19 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
     if (this.isAMandatoryComponent()) {
-      if ( this.clickInsideTheDocument && this.validateFormUploadedDocument() && !this.isUpLoadingAFile()) {
+      if (this.clickInsideTheDocument && this.validateFormUploadedDocument() && !this.isUpLoadingAFile()) {
         this.displayFileUploadMessages(WriteDocumentFieldComponent.UPLOAD_ERROR_FILE_REQUIRED);
       }
     }
   }
 
-  fileValidationsOnTab () {
+  fileValidationsOnTab() {
```
Comment (rayliangatos) [inline src/shared/components/palette/document/write-document-field.component.ts:132]: this has no modifier and return type

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 6]
```diff
@@ -100,19 +120,19 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
     }
   }
 
-  fileValidations () {
+  fileValidations() {
```
Comment (rayliangatos) [inline src/shared/components/palette/document/write-document-field.component.ts:123]: this has no modifier and return type

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 7]
```diff
@@ -100,19 +120,19 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
     }
   }
```
Comment (rayliangatos) [inline src/shared/components/palette/document/write-document-field.component.ts:121]: all these methods has not modifier and return types, could we add them please

**Review by connorpgpmcelroy — DISMISSED** [review]
Just a few very minor changes I noticed could be made. Apart from that, the code looks good to go.

**Review by kuda1992 — DISMISSED** [review]
Seems all fine

File: src/app.config.ts (hunk 2 above)
Comment (adnanakgun) [reply src/app.config.ts:8]: That sets the default value, hence making the change a non-breaking change

**Review by rayliangatos — APPROVED** [review]

**Review by udaydenduluri33 — APPROVED** [review]
Can see changes asked by review, thanks

**Comment by udaydenduluri33** [comment]
@adnanakgun  @RiteshHMCTS  - I am approving on behalf of @connorpgpmcelroy  who is on Holiday today

**Pull request merged**